pallets/marketplace/src/lib.rs

```diff
@@ -0,0 +1,486 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![allow(clippy::upper_case_acronyms)]
+
+use frame_support::{
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Currency, ExistenceRequirement, NamedReservableCurrency, OriginTrait},
+	transactional,
+};
+use frame_system::{ensure_signed, RawOrigin};
+use sp_runtime::{
+	traits::{Saturating, StaticLookup},
+	Percent,
+};
+
+use types::{Offer, TokenInfo};
+use weights::WeightInfo;
+
+use pallet_nft::{types::ClassType, Instances};
+use primitives::ReserveIdentifier;
+
+mod benchmarking;
+pub mod types;
+pub mod weights;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub type BalanceOf<T> =
+	<<T as pallet_nft::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub type TokenInfoOf<T> = TokenInfo<BalanceOf<T>>;
+pub type OfferOf<T> =
+	Offer<<T as frame_system::Config>::AccountId, BalanceOf<T>, <T as frame_system::Config>::BlockNumber>;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::OriginFor;
+
+	pub const RESERVE_ID: ReserveIdentifier = ReserveIdentifier::Marketplace;
```

Comment:
also probably it does not need to be pub.

i would also add a brief comment what's the purpose of this.

PErhaps - you can make this as config option ?

Reply:
removed pub and added a comment